Code/C++/unique_permutaions_of_string.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+A string is given we have to generate all its possible permutations.
+A permutation of the string is defined as a word that contains all the characters of the original string,
+but in any order. The output should contain only unique words in lexicographically sorted order.
+
+*/
+
+#include <iostream>
+#include<string>
+#include<set>
+using namespace std;
+
+//function to generate all premutations
+void permute(char *inp, int cur_Index, set<string> &ans) {
+	//base case (end of string i.e. all palces filled)
+	if (inp[cur_Index] == '\0') {
+		ans.insert(string(inp));
+		return;
+	}
+
+	//recusrcive case
+	for (int j = cur_Index; inp[j] != '\0'; ++j)
+	{
+		//we will place every available charater at current_Index
+		swap(inp[cur_Index], inp[j]);
+
+		//solving sub problem
+		permute(inp, cur_Index + 1, ans);
+
+		//undo the changes made
+		swap(inp[cur_Index], inp[j]);
+	}
+}
+
+
+int main()
+{
+
+	// all the words will be stored in set
+	//because set keeps only unique elements and stores elements in sorted order
+	//hence iterating on set will give all unique words in lexicographically sorted order
+	set<string> ans;
+	ans.clear();
```

Comment:
why is ans.clear() needed?

Reply:
it's my habit to always clear to avoid unexpected things in case of global declaration, but it was needed there. I will remove this and make other changes